Adds a known issue to 4.17 release notes (#82266)
openshift/openshift-docs: release_notes/ocp-4-17-release-notes.adoc

* The DNF package manager included in {op-system-first} images cannot be used at runtime, because DNF relies on additional packages to access entitled nodes in a cluster that are under a Red Hat subscription. As a workaround, use the `rpm-ostree` command instead. (link:https://issues.redhat.com/browse/OCPBUGS-35247[*OCPBUGS-35247*])

> * Due to a change in storage account naming in {product-title} {product-version}, the Azure File Container Storage Interface (CSI) driver now alphabetically matches storage accounts of the Image Registry Operator. With this change, there is a known issue where the Azure File CSI driver does not work properly when the image registry is configured as private. As a result, the image registry fails to mount volumes because it does not have a worker subnet added to allowed networks for the storage account that it tries to use.

Reviewer: ```suggestion
* Due to a change in storage account naming in {product-title} {product-version}, the Azure File Container Storage Interface (CSI) driver now alphabetically matches storage account of the Image Registry Operator. With this change, there is a known issue where the Azure File CSI driver fails to mount all volumes when image registry is configured as private. The mount failures occur because the CSI driver tries to use the storage account of the Image Registry Operator which is not configured to allow connections from worker subnet.
```

+

> As a temporary workaround, image registries should not be configured as private when using the Azure File CSI driver. This is a known issue and will be fixed in a future version of {product-title}. (link:https://issues.redhat.com/browse/OCPBUGS-42308[*OCPBUGS-42308*])

Reviewer 2: s/image registries/image registry/g

Reviewer 2: Hi @stevsmit @RomanBednar, IMO, we are better to add another workaround too, like https://issues.redhat.com/browse/OCPBUGS-38922 mentioned `using storageclass with networkEndpointType: privateEndpoint or specify another storage account` , in case that custom have to use private image registry.

Reviewer: @xiuwang I did consider providing this workaround already and I strongly suggest we don't do that. CSI storage with private endpoints is untested for now and we can't tell for sure if that would not cause any issues in the future during upgrades for example. That's why we did not introduce any code to change the storage class in this way, so suggesting that as a workaround officially would have similar effect.

We might consider adding this as a feature in next releases but the priority is not assessed yet.

Reviewer 2: @RomanBednar Thanks for the explaining, that makes sense. I didn't consider upgrade such situation.
